z3rro/src/prover.rs

```diff
@@ -69,6 +69,13 @@ impl<'ctx> Prover<'ctx> {
         set_solver_timeout(&self.solver, duration);
     }
 
+    pub fn enforce_ematching(&mut self) {
+        let mut params = Params::new(self.solver.get_context());
+        params.set_bool("auto-config", false);
+        params.set_bool("smt.mbqi", false);
```

Comment:
Maybe also add this to the Rustdoc documentation of the `Fuel` type or somewhere that preventing instantiations by the SMT solver of the fuel type is necessary and that disabling MBQI accomplishes that.